core/src/main/java/org/dozer/util/RuntimeUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2005-2017 Dozer Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dozer.util;
+
+import org.dozer.osgi.Activator;
+
+/**
+ * Utility class to provide various information about runtime in which Dozer is executed.
+ */
+public final class RuntimeUtils {
+
+    private RuntimeUtils() {
+    }
+
+    private static Boolean isOSGi;
+
+    /**
+     * Returns {@code true} if Dozer is executed in OSGi container.
+     *
+     * @return if Dozer is in OSGi container.
+     */
+    public static boolean isOSGi() {
+        if (isOSGi == null) {
+            try {
```

Comment:
Actually, we do cache the result. Field `isOSGi` is calculated once and then it is returned by the method without re-calculation. 

Regarding OSGi in classpath, this was addressed and that's also the reason to do two-step verification:
* first we check if OSGi is in classpath:
```
Class.forName("org.osgi.framework.Bundle")
```
* if it is, it may be put there by users, so we check if our bundle activator is initiated:
```
Activator.getBundle() != null
``` 

So far to me it seems to cover possible case.